StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1209UsingAliasDirectivesMustBePlacedAfterOtherUsingDirectives.cs

```diff
@@ -47,64 +51,67 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            context.RegisterSyntaxNodeActionHonorExclusions(this.HandleUsingDirectiveSyntax, SyntaxKind.UsingDirective);
+            context.RegisterSyntaxNodeActionHonorExclusions(this.HandleCompilationUnit, SyntaxKind.CompilationUnit);
+            context.RegisterSyntaxNodeActionHonorExclusions(this.HandleNamespaceDeclaration, SyntaxKind.NamespaceDeclaration);
         }
 
-        private void HandleUsingDirectiveSyntax(SyntaxNodeAnalysisContext context)
+        private void HandleCompilationUnit(SyntaxNodeAnalysisContext context)
         {
-            UsingDirectiveSyntax syntax = context.Node as UsingDirectiveSyntax;
-            if (syntax.Alias == null)
-            {
-                return;
-            }
+            var compilationUnit = context.Node as CompilationUnitSyntax;
 
-            CompilationUnitSyntax compilationUnit = syntax.Parent as CompilationUnitSyntax;
-            SyntaxList<UsingDirectiveSyntax>? usingDirectives = compilationUnit?.Usings;
-            if (!usingDirectives.HasValue)
-            {
-                NamespaceDeclarationSyntax namespaceDeclaration = syntax.Parent as NamespaceDeclarationSyntax;
-                usingDirectives = namespaceDeclaration?.Usings;
-            }
+            ProcessUsingsAndReportDiagnostic(compilationUnit.Usings, context);
+        }
 
-            if (!usingDirectives.HasValue)
-            {
-                return;
-            }
+        private void HandleNamespaceDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var namespaceDeclaration = context.Node as NamespaceDeclarationSyntax;
+
+            ProcessUsingsAndReportDiagnostic(namespaceDeclaration.Usings, context);
+        }
+
+        private static void ProcessUsingsAndReportDiagnostic(SyntaxList<UsingDirectiveSyntax> usings, SyntaxNodeAnalysisContext context)
+        {
+            UsingDirectiveSyntax usingAliasDirectivesShouldBePlacedAfterThis = null;
+            var usingAliasDirectivesToReport = new Lazy<List<UsingDirectiveSyntax>>();
```

Comment:
I don't think its very useful to use Lazy<T> here. The collection is empty when created (so its cheap to construct) and it will be used almost immediately (if there is anything to report). In my opinion the benefits of using Lazy are outweighed here by the increased complexity (most developers I know will have to think twice when they see such a construction).